### ISSUE TITLE
Add Lie Algebra R^dim for the Special Orthogonal group (with point type: "vector") 

### ISSUE THREAD

**Issue by SaitejaUtpala** [geomstats/geomstats#1470]
## What?

In Geomstats, the Lie group class, and the classes that inherit from it, have an attribute `lie_algebra` that contains an object specifying the Lie algebra corresponding to the Lie group.

See for example the Lie algebra of the Special Orthogonal group (with points represented as matrices) here:
https://github.com/geomstats/geomstats/blob/6e169c3fdee5b17135063816a48722124ddd0eb9/geomstats/geometry/special_orthogonal.py#L47

However, the Lie algebra of the Special Orthogonal group with points represented as vectors (for n=2 or n=3) is not included. This Lie algebra should be R^dim where dim is the dimension of SO(n) equal to n(n+1)/2 and should be implemented as an object instantiated from the [class VectorSpace](https://github.com/geomstats/geomstats/blob/de63a69833e03576646dd944e95574af689eb6e9/geomstats/geometry/base.py#L14).

This makes the code snippet provided below fail.

This issue aims to add these Lie algebras [to the class _SpecialOrthogonalVectors](https://github.com/geomstats/geomstats/blob/6e169c3fdee5b17135063816a48722124ddd0eb9/geomstats/geometry/special_orthogonal.py#L286).

## Where?

The attribute lie_algebra should be added here:
https://github.com/geomstats/geomstats/blob/6e169c3fdee5b17135063816a48722124ddd0eb9/geomstats/geometry/special_orthogonal.py#L286

## How?

- [ ] Reproduce the bug by running the code snippet below on your local machine
- [ ] Add the attribute lie_algebra [to the class _SpecialOrthogonalVectors](https://github.com/geomstats/geomstats/blob/6e169c3fdee5b17135063816a48722124ddd0eb9/geomstats/geometry/special_orthogonal.py#L286), e.g. through `self.lie_algebra(VectorSpace(dim=self.dim))`
- [ ] Make sure that the code snippet now runs
- [ ] Unskip the unit-tests that were skipping testing the Lie algebra, [by removing this line](https://github.com/geomstats/geomstats/blob/6e169c3fdee5b17135063816a48722124ddd0eb9/tests/tests_geomstats/test_special_orthogonal.py#L21).
- [ ] Make sure that the unit-tests run with the un-skipped unit-tests: `pytest tests/tests_geomstats/test_special_orthogonal.py`

### Additional information

The following code snippet currently fails. This issue should make it run.

```
from geomstats.geometry.special_orthogonal import SpecialOrthogonal
print(SpecialOrthogonal(2, 'vector').lie_algebra)
```

**Comment by juanitorduz** [comment]
Hi! I just got to know the library and would love to take a "good first issue" to contribute. I am interested in working on this issue. Is it's open to take?

**Comment by luisfpereira** [comment]
Thanks for identifying this was already solved @juanitorduz (in fact, commit c6d808b adds `self` as lie algebra of this space). Looking forward for your future contributions!
